sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -240,6 +240,10 @@ statement
     | MSCK REPAIR TABLE multipartIdentifier                            #repairTable
     | op=(ADD | LIST) identifier (STRING | .*?)                        #manageResource
     | SET ROLE .*?                                                     #failNativeCommand
+    | SET TIME ZONE ALL                                                #listTimeZones
+    | SET TIME ZONE interval                                           #setTimeZone
+    | SET TIME ZONE timezone=(STRING | LOCAL)                          #setTimeZone
+    | SET TIME ZONE .*?                                                #setTimeZone
```

Comment:
so we add this only for better parser message?

Reply:
this is used to fail invalid set time zone syntax explicitly, cuz' now we support 

```sql
spark-sql (default)> set time zone abcd;
key	value
time zone abcd	<undefined>
```